### ISSUE TITLE
ENH: upgrade votable units to use v1.1

### ISSUE THREAD

**Issue by bsipocz** [astropy/astropy#17242]
### What is the problem this feature will solve?

While browsing the code for the #17232 review I noticed that we are still using v1.0 for VOUnits. There has been some updates to it a year ago, so we should update the code to use v1.1.

Changes summary from the standard:

```
D.1  Changes from REC-1.0 to REC-1.1

    In FITS 4.0[ (FITS Working Group, 2018)](https://www.ivoa.net/documents/VOUnits/20231215/REC-VOUnits-1.1.html#fits4), the preferred symbol for the year is a, not yr (see Table [2](https://www.ivoa.net/documents/VOUnits/20231215/REC-VOUnits-1.1.html#tabx:knownunits)).

    The VOUnits syntax now accepts 1 as positively indicating a dimensionless unit string, and the text no longer recommends that the empty string `positively indicates that the corresponding quantity is dimensionless'. An application is free to interpret the empty string as indicating a dimensionless quantity, that the units are unknown, or that this is a parsing error (see Sect. [2.2](https://www.ivoa.net/documents/VOUnits/20231215/REC-VOUnits-1.1.html#sec:parsing-components)).

    Function operands can now contain a scale-factor, so that, for example, log(10**6Hz) is an alternative to log(MHz), though the latter is preferred (see Sect. [2.9](https://www.ivoa.net/documents/VOUnits/20231215/REC-VOUnits-1.1.html#sec:math)).

    Added: brief rationale for the syntax of the VOUFLOAT terminal (see Appx. [C.4](https://www.ivoa.net/documents/VOUnits/20231215/REC-VOUnits-1.1.html#appx:vougrammar)).

    Added: new SI prefixes `r', `q', `R', and `Q' (see Sect. [2.6](https://www.ivoa.net/documents/VOUnits/20231215/REC-VOUnits-1.1.html#sec:scalingprefixes)).

    Added: a new VOUnits syntax summary section, for impatient readers (see Sect. [1.5](https://www.ivoa.net/documents/VOUnits/20231215/REC-VOUnits-1.1.html#sec:cribsheet)).

    Various minor adjustments and clarifications to the text.

    Document management: ported to ivoatex, streamlined use cases, several editorial changes.

Problem reports for the REC-1.0 document were tracked on GitHub, which is where the document was managed during redrafting. The changes were discussed on that site, on IVOA mailing lists, and at Interops.
```

**Comment by bsipocz** [comment]
ps @pllim - the template is really annoying, can we have a blank one, too?

**Comment by mhvk** [comment]
Thanks for putting up a separate issue. Related is #17230 about dimensionless units, and a question that was raised there about how to signal what standards version one wants to use - https://github.com/astropy/astropy/pull/17230#issuecomment-2429650460

This may need some thought: the basic implementation would surely be that by default we simply accept the unit strings given according to any version, and write as we should according to the latest version.

The question is whether we need to allow choice, e.g., to parse strictly according to a given version, or  to write according to one. 

FWIW, I think we do *not* have to implement strict version-dependent parsing - I do not see why astropy would have that role, that should be something provided by the VO standard itself if it is deemed important. This of course unless the meaning of a given string differs between versions - I really hope not!

For writing, I'm less sure, but it would seem to depend again on whether any new version has backward incompatibility issues with earlier ones. I think if something written with the latest standard will always be interpreted correctly if read according to an earlier one, then there should be no need to ever do anything but write according to the latest version.

**Comment by bsipocz** [comment]
I think that most of the standards are interpreted as use latest and greatest rather than a pick your version. So I don't think we need to provide anything for the older ones, besides clear documentation which one we use. 

But @tomdonaldson or @msdemlei will know this correctly.

**Comment by mhvk** [comment]
Makes sense - as long as we can still interpret files using the older standards correctly - but standards should be backwards-compatible in that sense.

**Comment by msdemlei** [comment]
On Thu, Oct 24, 2024 at 09:43:26AM -0700, Marten van Kerkwijk wrote:
> Makes sense - as long as we can still interpret files using the
> older standards correctly - but standards should be
> backwards-compatible in that sense.

Exactly.  It would be a standards bug deserving a fix in the standard
text (or at least some sort of mitigating weaseling in case it's been
a bug before) if this were different.
